common/static/common/js/discussion/discussion_router.js

```diff
@@ -129,9 +128,7 @@
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
                 var thread;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                return this.showThread(thread);
```

Comment:
@bradenmacdonald The difference between `navigateToThread` and `showThread` is that `navigateToThread` has access to an up-to-date thread collection, so doing the lookup inside that method will succeed, and `showThread` doesn't. I don't know why `showThread` only has access to stale data, but I think this issue might be related to the `Uncaught Error: Backbone.history has already been started` error that comes up in the console. From the Backbone.js documentation on `Backbone.Router`:

> During page load, after your application has finished creating all of its routers, be sure to call `Backbone.history.start()` or `Backbone.history.start({pushState: true})` to route the initial URL.

`navigateToThread` does not act as a handler for a route (it is not mentioned in `DiscussionRouter.prototype.routes`), so it wouldn't be affected by any problems resulting from the fact that the `Backbone.history.start` call fails with the error mentioned above. So far, this is the only hint I found that might explain why doing the lookup inside `showThread` produces a different result than doing the lookup inside `navigateToThread`.

> Also, it seems like with this change, the URL is not going to get updated when the user clicks on the thread.

I pushed an update that allows the URL to change when clicking on a specific thread: https://github.com/edx-solutions/edx-platform/pull/741/commits/ecb6ce70f662dced72897bba6175a59a6f411fde.

However, since you mentioned [here](https://github.com/edx-solutions/edx-platform/pull/741#issuecomment-242170680) that we shouldn't implement too many fixes until [YONK-372](https://openedx.atlassian.net/browse/YONK-372) is completed, and the existing changes are more of a workaround, I'd also be fine with *not* trying to address the problem of not being able to show a newly created thread right away in the context of this PR. Let me know.